src/meshes/t8code_mesh.jl

```diff
@@ -100,6 +100,125 @@ function Base.show(io::IO, ::MIME"text/plain", mesh::T8codeMesh)
     end
 end
 
+"""
+    T8codeMesh(forest, boundary_names; polydeg, mapping=identity)
```

Comment:
You have a point here. As of now there is no convenient routine to get the dimension of a `forest`. Actually the `forest` does not know or even care about dimensions. Geometry related stuff is stored in the `cmesh`. But I opened a PR https://github.com/DLR-AMR/t8code/pull/1057 which adds a dimension getter for the `cmesh`. With that we can simplify the function signature here. I'll do that as soon as the getter is available in `T8code.jl`. This is when the next t8code release gets out.

Reply:
Ok, that's nice. So do you think we can proceed as follows?
1. Add the type parameters to the docstring as I suggested above.
2. Merge this PR.
3. Merge your PR in t8code and create a new release of t8code.
4. Create a new T8code.jl version including the new t8code version.
5. Create a new PR in Trixi.jl that updates the T8code.jl compat and creates a new convenience constructor that does not need the `NDIMS` type parameter explicitly because it is obtained from the new t8code function.

Reply:
That's the plan!